addons/udes_stock/models/stock_picking_batch.py

```
@@ -1017,21 +1033,45 @@ def reserve_pallet(self, pallet_name):
         Raises a ValidationError if the pallet is already reserved for another
         batch.
         """
+        Picking = self.env['stock.picking']
         PickingBatch = self.env['stock.picking.batch']
 
         self.ensure_one()
 
-        conflicting_batch = PickingBatch.search([
-            ('id', '!=', self.id),
-            ('state', '=', 'in_progress'),
-            ('u_last_reserved_pallet_name', '=', pallet_name),
-        ])
-        if conflicting_batch:
-            raise ValidationError(
-                _("This pallet is already being used for batch %s.") %
-                conflicting_batch[0].name)
+        reserve_pallet_per_picking = self.picking_type_ids.filtered('u_reserve_pallet_per_picking')
+        if reserve_pallet_per_picking:
+            if not picking:
+                raise ValidationError('A picking must be specified if pallets are reserved per picking.')
+        if reserve_pallet_per_picking and picking and not self.picking_ids & picking:
+            return
```

Do we want to raise an error in this case?